File: anylang/go/leetcode/1486.数组异或操作.go
package leetcode

import "fmt"

func xorOperation(n int, start int) int {
	if n == 0 {
		return 0
	}

	rst := start
	for i := 1; i < n; i++ {
		rst ^= start + 2*i
	}

	return rst
}

func XorOperation() {
	fmt.Println("xorOperation(5, 0): ", xorOperation(5, 0))
	fmt.Println("xorOperation(4, 3): ", xorOperation(4, 3))
	fmt.Println("xorOperation(1, 7): ", xorOperation(1, 7))
}

File: anylang/c++/1486.数组异或操作.cpp
#include <stdio.h>
#include <iostream>

class Solution {
   public:
    int xorOperation(int n, int start) {
        if (n == 0) {
            return 0;
        }

        int rst = start;
        for (int i = 1; i < n; i++) {
            rst ^= start + 2 * i;
        }

        return rst;
    }
};

int main() {
    // printf("xorOperation(5, 0): %d", xorOperation(5, 0));
    Solution s;
    std::cout << "xorOperation(5, 0): " << s.xorOperation(5, 0) << std::endl;
    return 0;
}
